### PR TITLE
DB-10961 Get rid of the NodeFactory part 4

### PR DESCRIPTION
**Pull request by martinrupp**
iterative PR to get rid of NodeFactory / getNode

### CODE REVIEW

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0

**Comment by cloudspliceci** [comment]
TEST SUCCEEDED +1

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0

**Comment by cloudspliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/3160/

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by cloudspliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/3190/

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by cloudspliceci** [comment]
TEST SUCCEEDED +1

**Review by dgomezferro — CHANGES_REQUESTED** [review]
Change looks good, I only have a comment regarding the `.equals()` implementation.

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/AllResultColumn.java [hunk 1]
```diff
@@ -43,62 +44,67 @@
...
+        return new AllResultColumn(tableName, getContextManager());
+    }
 
     public TableName getTableNameObject() {
         return tableName;
     }
+
+    public boolean equals(Object o){
+        return this == o;
+    }
```
[the remark this answers is not in the document]
Comment (martinrupp) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/AllResultColumn.java:109]: yeah this is weird... SpotBugs complaint about missing equals function, but i don't think we need this. will remove again

**Review by dgomezferro — APPROVED** [review]
Thanks for the changes Martin, looks good to me!

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by cloudspliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/3206/

**Review by jyuanca — APPROVED** [review]

**Comment by cloudspliceci** [comment]
TEST SUCCEEDED +1

**Review by carolp-503 — APPROVED** [review]
+1

**Pull request merged**